components/org.wso2.carbon.identity.api.user.recovery/src/main/java/org/wso2/carbon/identity/recovery/endpoint/impl/RecoverPasswordApiServiceImpl.java

```diff
@@ -85,6 +85,10 @@ public Response recoverPasswordPost(RecoveryInitiatingRequestDTO recoveryInitiat
             if (LOG.isDebugEnabled()) {
                 LOG.debug("Client Error while sending recovery notification ", e);
             }
+            if (StringUtils.equals(IdentityRecoveryConstants.ErrorMessages.ERROR_CODE_FEDERATED_USER.getCode(),
+                    e.getErrorCode())) {
```

Comment:
```suggestion
            if (IdentityRecoveryConstants.ErrorMessages.ERROR_CODE_FEDERATED_USER.getCode().equals(
                    e.getErrorCode())) {
```

Reply:
resolved in https://github.com/wso2-extensions/identity-governance/pull/670/commits/a6c0423c0801bb6708adfd14cfc30c76f7bf2014